src/test/java/com/microsoft/sqlserver/jdbc/exception/ExceptionTest.java

```diff
@@ -93,6 +95,50 @@ public void testSocketTimeoutExceptionCause() throws Exception {
         }
     }
 
+    @Test
+    public void testResultSetErrorSearch() throws Exception {
+        String tableName = AbstractSQLGenerator.escapeIdentifier(RandomUtil.getIdentifier("ISSUE659TABLE"));
+        String procName = AbstractSQLGenerator.escapeIdentifier(RandomUtil.getIdentifier("ISSUE659PROC"));
+        String expectedException = "Error occured during the insert";
+        int outputValue = 5;
+        String createTableSql = "CREATE TABLE " + tableName + "(FIELD1 VARCHAR (255) NOT NULL);";
+        String createProcSql = "CREATE PROCEDURE " + procName
+                + " AS BEGIN TRANSACTION\n BEGIN TRY SET NOCOUNT ON; INSERT INTO " + tableName + " (FIELD1) OUTPUT "
+                + outputValue + " VALUES ('test'); INSERT INTO" + tableName
+                + " (FIELD1) VALUES (NULL); COMMIT TRANSACTION; END TRY BEGIN CATCH DECLARE @errorMessage NVARCHAR(4000) = ERROR_MESSAGE(); ROLLBACK TRANSACTION; RAISERROR('"
+                + expectedException + ": %s', 16, 1, @errorMessage); END CATCH;";
+        String execProcSql = "EXECUTE " + procName;
+
+        try (Connection conn = PrepUtil.getConnection(connectionString); Statement stmt = conn.createStatement()) {
```

Comment:
Prefer to use `getConnection()` method available from super class if no custom connection string needs to be passed.